DISCIPLINA: Gestão do Conhecimento	 PROFESSOR(A): Alex Salgado
PERÍODO: 7o. TURNO: noite AVALIAÇÃO:

ALUNO(A): Thiago Guimarães de Salles

GRAU: VISTO DO PROFESSOR:


# Questão 1 - valor (1,0)


*** Crie um repositório no Github e adicione este arquivo no mesmo


1.1 - Você deve escolher uma base de dados aberta que seja possível treinar um modelo de aprendizagem de máquina Supervisionado de Classificação (assim como o exemplo da Iris e Cancer). A partir desta base de dados, você vai treinar o modelo, fazer previsões e calcular a acurácia do mesmo.
Obs.: Não utilizar nenhuma base de dados usada em aula ou nos trabalhos.


Base de dados contendo a classificação de tipos de vidros

1.2 - Após tratada, criar uma pasta no seu GoogleDrive e disponibilizar o link do arquivo de sua base (csv, xlsx, etc) para ser importado no seu código.

https://drive.google.com/open?id=1PV_3JHsXIeNXLZbTMLP7_3vV4inO2TFM

## Criar um código usando o Jupyter Notebook e responder (através de código) às seguintes questões:


# Questão 2 - valor (1,0)

2.1 - Importar os modulos python para machine learn e carregar o arquivo 

** Sugestão se for do tipo xlsx, usar o read_excel do pandas
import pandas as pd
dt = pd.read_excel("meuarquivo.xlsx")


In [0]:
from sklearn.neighbors import KNeighborsClassifier
from sklearn.linear_model import LogisticRegression
from sklearn import metrics
from sklearn.model_selection import train_test_split
import matplotlib.pyplot as plt
import pandas as pd

data = pd.read_csv("glass.csv")
X = data[['RI', 'Na', 'Mg', 'Al', 'Si', 'K', 'Ca', 'Ba', 'Fe']]
y = data['Type']

2.2 - Exibir os primeiros registros desta tabela.


In [24]:
data.head()

RI     Na    Mg    Al     Si     K    Ca   Ba   Fe  Type
0  1.52101  13.64  4.49  1.10  71.78  0.06  8.75  0.0  0.0     1
1  1.51761  13.89  3.60  1.36  72.73  0.48  7.83  0.0  0.0     1
2  1.51618  13.53  3.55  1.54  72.99  0.39  7.78  0.0  0.0     1
3  1.51766  13.21  3.69  1.29  72.61  0.57  8.22  0.0  0.0     1
4  1.51742  13.27  3.62  1.24  73.08  0.55  8.07  0.0  0.0     1

2.3 - De que se trata esse banco de dados? e que tipo de previsão pode ser feito com ele?



Este é um conjunto de dados de identificação de vidro da UCI. Através dessa base de dados conseguimos identificar qual é o tipo de vidro, baseado em seus atributos.

# Questão 3 - valor (0,5)


Utilizando as terminologias de Machine Learning(features e observações):



3.1 - Quantas "features" têm nessa base de dados?

In [32]:
print(X.shape)
print(y.shape)

(214, 9)
(214,)


9 features

3.2 - Quantas observações têm nessa base de dados?


214 observações

# Questão 4 - valor (1,0)


4.1 - Faça uma previsão, usando o algoritmo de LogisticRegression.


In [26]:
logreg = LogisticRegression()
logreg.fit(X, y)
logreg.predict(X)

/usr/local/lib/python3.6/dist-packages/sklearn/linear_model/_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)


array([1, 2, 2, 1, 1, 2, 1, 1, 2, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 2, 2, 1,
       1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1,
       1, 2, 1, 1, 1, 2, 1, 2, 2, 2, 2, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1,
       1, 1, 1, 1, 2, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
       2, 2, 1, 2, 2, 2, 2, 2, 1, 1, 2, 2, 2, 2, 1, 1, 2, 5, 5, 2, 6, 6,
       2, 2, 2, 1, 1, 1, 1, 2, 2, 2, 1, 2, 2, 2, 1, 2, 1, 2, 2, 2, 2, 2,
       1, 2, 1, 1, 1, 2, 2, 2, 2, 1, 2, 2, 2, 1, 1, 2, 2, 1, 2, 1, 1, 2,
       1, 1, 2, 1, 2, 2, 2, 1, 1, 7, 2, 2, 2, 5, 5, 5, 5, 5, 5, 2, 2, 2,
       2, 2, 2, 2, 7, 6, 7, 6, 6, 7, 7, 2, 2, 7, 7, 7, 7, 7, 7, 7, 7, 7,
       7, 7, 7, 5, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7])

# Questão 5 - valor (1,5)


Usando o método de avaliação de acurácia (Treinar e testar na base de dados inteira/Train test entire model), Calcular a acurácia de cada um dos 3 métodos abaixo.

4.1 - Acurácia usando o algoritmo de KNN (com 1 vizinho, k=1)


In [27]:
y_pred = logreg.predict(X)
len(y_pred)
knn = KNeighborsClassifier(n_neighbors=1)
knn.fit(X, y)
y_pred = knn.predict(X)
print(metrics.accuracy_score(y, y_pred))

1.0


4.2 - Acurácia usando o algoritmo de KNN (com 5 vizinho, k=5)


In [28]:
y_pred = logreg.predict(X)
len(y_pred)
knn = KNeighborsClassifier(n_neighbors=5)
knn.fit(X, y)
y_pred = knn.predict(X)
print(metrics.accuracy_score(y, y_pred))

0.7616822429906542


4.3 - Acurácia usando o algoritmo de LogisticRegression


In [29]:
y_pred = logreg.predict(X)
len(y_pred)
print(metrics.accuracy_score(y, y_pred))

0.6495327102803738


4.4 - De acordo com seus resultados anteriores, qual dos 3 métodos é mais eficiente?


KNN = 1

In [0]:
*** Envie o link do seu Github com essa resposta no Classroom
